File: $_algotsu-kaizen/problemo-set/minimum-add-to-make-parentheses-valid.c

int minAddToMakeValid(char* s) {
    int open = 0b0, close = 0b0, len = 0b0, c = '\0';
    while (c = *(s + len++)) {
        open += c == '(' ? 1 : -1;
        if (open < 0) {
            close += 1;
            open = 0;
        }
    }
    return open + close;
}

File: $_algotsu-kaizen/problemo-set/minimum-add-to-make-parentheses-valid.rs

impl Solution {

    pub fn min_add_to_make_valid(s: String) -> i32 {
        let mut open = 0b0;
        let mut close = 0b0;
        for c in s.chars() {
            open += if c == '(' { 0b1 } else { -0b1 };
            if open < 0b0 {
                close += 0b1;
                open = 0b0;
            }
        }
        return open + close;
    }
}
